### PR TITLE
Replace references to the old Buttonweavers wiki at Wikia/Fandom with references to the new buttonmen.wiki wiki.

### LINKED ISSUE #2914: Change links to wiki to redirect to new site
A number of our members have migrated the Fandom Button Men wiki (https://buttonmen.fandom.com/wiki/Main_Page) and Buttonweavers wiki (https://buttonweavers.fandom.com/wiki/Main_Page) to a new wiki ~~on Miraheze (https://buttonmen.miraheze.org/wiki/Main_Page)~~ at https://buttonmen.wiki/wiki/Main_Page.

We should update both the direct wiki links and the help to reflect this migration.

### PR DESCRIPTION
**Pull request by irilyth**
Fixes #2914.

### CODE REVIEW

**Comment by blackshadowshade** [comment]
There is at least one other place that needs to be changed in the Help documentation.

**Comment by blackshadowshade** [comment]
I was looking at the new wiki today, and it seems like the migration is incomplete? The Fanatics are certainly incomplete.

Can anyone weigh in on whether we still need the links from the old wikis?

**Comment by irilyth** [comment]
Hm, I did a `find | grep` for `buttonweavers.fandom.com` and didn't come up with anything else. Something else in [this Help.js file](https://github.com/buttonmen-dev/buttonmen/blob/master/src/ui/js/Help.js)? Or some other place?

**Comment by blackshadowshade** [comment]
Yup, the Help.js file is what you're looking for.

**Comment by blackshadowshade** [comment]
I guess I didn't phrase that correctly. I was under the impression that both wikis had been merged into one. Is that not correct?

**Comment by irilyth** [comment]
> I was looking at the new wiki today, and it seems like the migration is incomplete? The Fanatics are certainly incomplete.

Do you have an example? I did a quick search for "fanatics" on the old wiki, and didn't see an obvious page of Fanatics or anything.

> Can anyone weigh in on whether we still need the links from the old wikis?

The only place those links show up is in forum code, right? I'm pretty willing to assert that no one's created a wiki link in any forum post in quite some time; the old wiki is 100% dead as far as I can tell, I don't think anyone's edited any pages there in a year or two (although it's hard to tell, since the recently updated pages list is empty, but that itself is probably a sign).

**Comment by irilyth** [comment]
> I guess I didn't phrase that correctly. I was under the impression that both wikis had been merged into one. Is that not correct?

Oh, you mean the buttonmen.fandom.com wiki? I have no idea if it's been merged into buttonmen.wiki, I was never particularly involved with it; but the forum links never pointed there anyway, so it's not really relevant to #2914 or this PR in any case.

**Comment by craw-daddy** [comment]
As far as I am aware, there were not individual pages for the Fanatics in the old wiki.  Of course, there were not individual pages for any buttons in the okd wiki.  Yes, the main Fanatics set page still needs editing, but that can be a work 8n progress, and maybe we actually get someother people to help out there.  I basically made 98% of the individual button pages myself.

TLDR, I think the migration is basically complete.  There is some work left to be done, but that’s just further improvements.

**Comment by blackshadowshade** [comment]
Yup, irilyth, you got my gist now, sorry for my imprecision.

I seem to remember that TheOrgg had collected stories about Fanatics on the old wiki, and I couldn't find those on the new wiki. Is that because they were on the buttonweavers wiki and not the Button Men wiki (or vice versa)?

**Comment by irilyth** [comment]
> I seem to remember that TheOrgg had collected stories about Fanatics on the old wiki, and I couldn't find those on the new wiki. Is that because they were on the buttonweavers wiki and not the Button Men wiki (or vice versa)?

Could be; I don't remember the history of buttonmen.fandom.com and how it relates to buttonmen.wiki.

The particular problem I'm trying to solve is that forum links like `[wiki=foo]` currently point to buttonweavers.fandom.com, and they should instead point to buttonmen.wik, which is what #2914 is abouti. I'm not really interested in taking on a project like "improve the wiki situation in general".

**Comment by blackshadowshade** [comment]
I was just surprised that I still saw two wikis on the help page when I had expected one. If we still have two wikis though, that's fine.

Regarding CircleCI, I can't find the link to its run, although you noted on Discord that it indicates errors. Could you drop a link to the run (if you have it)?

**Comment by blackshadowshade** [comment]
Oh wait, found it: https://app.circleci.com/pipelines/github/irilyth/buttonmen?branch=2914_wiki_links

**Comment by blackshadowshade** [comment]
The error is this:

Either git or ssh (required by git to clone through SSH) is not installed in the image. Falling back to CircleCI's native git client but the behavior may be different from official git. If this is an issue, please use an image that has official git and ssh installed.
Cloning git repository

error cloning repository: ssh: handshake failed: ssh: unable to authenticate, attempted methods [none publickey], no supported methods remain

**Comment by irilyth** [comment]
Hm, sure; I got three messages, with these links:

1. [build](https://email.circleci.com/c/eJw8jrtOwzAYhZ_mz5YqsRPbGTJQlYihDBUDhQW5viS_8KVynJbw9Kgq6nZ0dC6f7gXrRGH6mldciKZlpJj69mQqQ3grq0521lLWKkFVZWmlW9Z2usB-fHne74-H3_fjZ_hZD5f64_Vte9g8ZuqOC2iqc5yzdBuFSTlTOjNKtZYhZrRoUnmpS8E073jDhC2DdoQXrp9yPs9An4AMQIZ7VeFGRQ9kGCcgAyZ0a76p05JzDN4EIENNgA5L9l9K-rPEMQDdXWP6ti5eSyvRGQ2E3QLeaFw80J3xEt2_OcclKQN0d8dTMmMMReofZ9sR87ScoKkwWJPmdb4x_QUAAP__Addo-A)
2. [build](https://email.circleci.com/c/eJxEjj1P8zAUhX_NzZaqtlPHGTK8rypvSCAGRuQ61_EFfwTHISq_HlVFsB0dnY9nGpUcVIMj64-9Ut1pYI0fuUTHxaS4kJaZC_aqHxCRuRMznZWnhsb88LHwpz2pR7brF989v2HZD38rnAnojkteqwkHS8UGbAPOxl7blCs5wtJ-slbJqR_6TirXhq_EfRNGX-uygvgHXAPX96qlg80RuJ49cE2FwrXe1GWrNaeICbhmAoTeany1Ji6G5gTivOfy7kLeW2co4ARc3gIRJ9oiiDNGQ-HHXPNWLII43_GsqZRTU8bfs_8zVb9doDtScljW63pj-g4AAP__Je5pQg)
3. [build](https://email.circleci.com/c/eJw8jjtrwzAUhX_N9eZgSX7Ig4eG1FAKKZQO7VQU6cq-VA8jyw3Jry8hJdvhcB6fGWTbywIH1lWdlHUvu2IeDIqmsgYbIXjfMNYyy5Q1TMuuR6VsQcP-4_nt63g9vlR783l95dP0bpbz7jHTdnUDdbXENSu305S0w9LhpPSlDDGTJUzlLytla7q-q1tpS3cNfC7cMOe8rCCegI_Ax3tV005HD3ycZuAjJXKXfFOnLecYPAbgI6tBjFv231r5RdEUQBzOMf1YF8-lVeTQAG9vAY-GNg_igF6R-zfXuCWNIA53PK0yxVCk4XG2nyjP2wnqioLFtF7WG9NfAAAA__9v8Wl9)

I'm not sure if that's enough to go on -- if it's not, lemme know and I can copy more of what was in the e-mails.

**Comment by blackshadowshade** [comment]
That error seems pretty odd, I can't imagine why CircleCI can't find either git or ssh.

**Comment by blackshadowshade** [comment]
On my most recent passing pull request, my "Checkout code" log looks like:

Either git or ssh (required by git to clone through SSH) is not installed in the image. Falling back to CircleCI's native git client but the behavior may be different from official git. If this is an issue, please use an image that has official git and ssh installed.
Cloning git repository
Enumerating objects: 44148, done.
Counting objects: 100% (1870/1870), done.
Compressing objects: 100% (1024/1024), done.
Total 44148 (delta 955), reused 1564 (delta 786), pack-reused 42278
Checking out branch
HEAD is now at 95d4fcd Yearly update of copyright year to 2024

**Comment by blackshadowshade** [comment]
Looks like there may be a problem associated with the deploy key, which I see here:

https://github.com/buttonmen-dev/buttonmen/settings/keys

**Comment by blackshadowshade** [comment]
My thought, @irilyth, when you have time, is that you may need to create or associate a deploy key.

See https://circleci.com/docs/github-integration/ and https://stackoverflow.com/questions/74004701/circle-ci-error-cloning-repository-repository-not-found

**Comment by irilyth** [comment]
Those docs don't seem to have step-by-step instructions, and I'm not sure where to put the various parts of this.

[At CircleCI](https://app.circleci.com/settings/project/github/irilyth/buttonmen/ssh), I see

> Deploy Key
> [irilyth/buttonmen deploy key](https://github.com/irilyth/buttonmen/settings/keys) PREFERRED
> SHA256:3k4wxnmqVluLAQS1UMBlS4VxCjt9Py0r18zEYhGyLW0

but when I click on that link, GitHub says "There are no deploy keys for this repository". So perhaps I need to get the public key from CircleCI and add it to GitHub? That `SHA256` business is a fingerprint, not the pubkey itself, and I don't see an obvious way to get it.

I tried adding the fingerprint at GitHub, but that didn't work, it says "Key is invalid. You must supply a key in OpenSSH public key format"

There's an "X" button which I suspect might delete the current CircleCI key; perhaps if I do that, it'll let me create a new one, and give me the pubkey at that time? I don't see an obvious way to add a new readonly key without deleting the current one. I can give that a try if no one has any better ideas -- the current one doesn't seem to be doing anything useful anyway, at least not at GitHub.

**Comment by irilyth** [comment]
Aha; I decided to just try it, so I deleted the existing deploy key, and added a new one, and it didn't show me the pubkey, but it automatically added it to GitHub. I wonder where the previous one went; maybe I got a notice about it not having been used in a while and deleted it, or something.

Anyway, back at CircleCI, I told it to re-run the pipeline, and that seems to have worked!

**Comment by cgolubi1** [comment]
I put up a dev site for this at https://2914-wiki-links.irilyth.dev.buttonweavers.com/ui/ --- i think we're still not merging things (i made an exception for the copyright year) until we have a path to deployment, but if someone says it looks good, i'll tag it ready-to-merge.

**Comment by blackshadowshade** [comment]
I've created a wiki link on the forum for UBFC (which was the example given) and that works.

The help links appear to work correctly. I've identified the thing that throws me off every time though: the title of the Buttonweavers wiki is "Buttonmen Wiki". That's not to be fixed in this pull request though.

**Comment by cgolubi1** [comment]
This is ready to merge and has been checked, so i'm going to tag it that way and tear down the dev site.

**Comment by cgolubi1** [comment]
Merge freeze is over, so let's merge this.
